integration-tests/eventbridge/nino_issue_credential/nino-issue-credential-publish.test.ts

```diff
@@ -0,0 +1,315 @@
+    expect(vcIssuedDetailType).toBe("VC_ISSUED");
+    expect(vcIssuedSource).toBe("review-hc.localdev.account.gov.uk");
+  });
+  it("should produce END and VC_ISSUED Events structure expected for the TxMA destination queue using target AuditEvent Step Function", async () => {
+    const startExecutionResult = await getExecutionResult("Bearer unhappy");
+    const txMaAuditEventTestQueueMessage = await getQueueMessages(
+      txMaAuditEventTestQueue.QueueUrl as string
+    );
+
+    const txMaPayload = txMaAuditEventTestQueueMessage.map(
+      (queueMessage) => JSON.parse(queueMessage.Body as string).detail
+    );
+
+    expect(startExecutionResult.output).toBeDefined();
+    const expected = [
+      {
+        component_id: "https://review-hc.dev.account.gov.uk",
+        event_name: "IPV_HMRC_RECORD_CHECK_CRI_END",
+        event_timestamp_ms: expect.any(Number),
+        timestamp: expect.any(Number),
+        user: {
+          govuk_signin_journey_id: "252561a2-c6ef-47e7-87ab-93891a2a6a41",
+          ip_address: "00.100.8.20",
+          persistent_session_id: "156714ef-f9df-48c2-ada8-540e7bce44f7",
+          session_id: "issue-credential-happy-publish",
+          user_id: "test",
+        },
+      },
+      {
+        component_id: "https://review-hc.dev.account.gov.uk",
+        event_name: "IPV_HMRC_RECORD_CHECK_CRI_VC_ISSUED",
+        event_timestamp_ms: expect.any(Number),
+        extensions: {
+          evidence: [
+            {
+              attemptNum: 2,
+              ciReasons: [
+                {
+                  ci: expect.any(String),
+                  reason: expect.any(String),
+                },
+              ],
+              failedCheckDetails: [
+                { checkMethod: "data", dataCheck: "record_check" },
+              ],
+              txn: expect.any(String),
+              type: "IdentityCheck",
+            },
+          ],
+        },
+        restricted: {
+          birthDate: [{ value: "1948-04-23" }],
+          name: [
+            {
+              nameParts: [
+                { type: "GivenName", value: "Jim" },
+                { type: "FamilyName", value: "Ferguson" },
+              ],
+            },
+          ],
+          socialSecurityRecord: [{ personalNumber: "AA000003D" }],
+        },
+        timestamp: expect.any(Number),
+        user: {
+          govuk_signin_journey_id: "252561a2-c6ef-47e7-87ab-93891a2a6a41",
+          ip_address: "00.100.8.20",
+          persistent_session_id: "156714ef-f9df-48c2-ada8-540e7bce44f7",
+          session_id: "issue-credential-happy-publish",
+          user_id: "test",
+        },
+      },
+    ];
+    expect(txMaPayload).toContainEqual(expected[0]);
+    expect(txMaPayload).toContainEqual(expected[1]);
+  });
+
+  const getExecutionResult = async (token: string) =>
+    executeStepFunction(output.NinoIssueCredentialStateMachineArn as string, {
+      bearerToken: token,
+    });
+
+  const getSessionItem = (
```

Comment:
Do you think this could be put somewhere where it can be reused?

Reply:
I have just tried to doing this but it ended up get more messy, I thought maybe making the session-item an interface and then we can have something consist for creating session-items but that ended up needing to change other parts of the system. Given this block is only used in 1 area of this test and we haven't got a need for re-use yet, maybe just leave it? 

Let me know what you think.

Reply:
Sure, this PR is already quite large so don't want to make it any bigger. Resolving 👍